src/chart/funnel/funnelLayout.ts

```diff
@@ -337,55 +427,156 @@ export default function funnelLayout(ecModel: GlobalModel, api: ExtensionAPI) {
+                but you set it as '${exitWidth}'`);
+            }
+        }
+
+        const showRate = seriesModel.get('showRate');
+        let firstVal: number;
+        const setLayoutPoints =
+            // The subsequent funnel shape modification will be done in this func.
+            // We don’t need to concern direction when we use this function to set points.
+            function (
+                index: number,
+                idx: number,
+                nextIdx: number,
+                sideLen: number,
+                pos: number
+            ): void {
+                if (dynamicHeight) {
+                    const start = getLinePointsBySize(pos, resSize / maxSize * viewSize);
+                    let end;
+
+                    if (index === indices.length - 1 && exitShape === 'rect') {
+                        end = getLinePointsBySize(pos + sideLen, resSize / maxSize * viewSize);
+                    }
+                    else {
+                        resSize += sort === 'ascending' ? sideLen : -sideLen;
+                        end = getLinePointsBySize(pos + sideLen, resSize / maxSize * viewSize);
                     }
+
+                    data.setItemLayout(idx, {
+                        points: start.concat(end.slice().reverse())
+                    });
+                    return;
                 }
+                else if (showRate) {
+                    const dataStart = getLinePoints(idx, pos);
+                    let dataEnd;
+                    if (exitWidth !== undefined && index === indices.length - 1) {
+                        const val = data.get(valueDim, idx) as number || 0;
+                        const itemSize = linearMap(val, [min, max], sizeExtent, true);
+                        const exitSize = itemSize * (exitWidth as number) / 100;
+                        dataEnd = getLinePointsBySize(pos + sideLen / 2, exitSize);
+                    }
+                    else {
+                        dataEnd = getLinePoints(idx, pos + sideLen / 2);
+                    }
+                    const rateStart = getLinePoints(idx, pos + sideLen / 2);
+                    const rateEnd = getLinePoints(nextIdx, pos + sideLen);
 
-                const start = getLinePoints(idx, y);
-                const end = getLinePoints(nextIdx, y + height);
+                    const val = data.get(valueDim, idx) as number || 0;
+                    const nextVal = data.get(valueDim, nextIdx) as number || 0;
+                    let rate: number | string = nextVal / val;
+                    rate = 'Rate ' + (rate * 100).toFixed(0) + '%';
+                    if (index === 0) {
+                        firstVal = val;
+                    }
+                    else if (index === indices.length - 1) {
+                        const lastVal = val;
+                        rate = lastVal / firstVal;
+                        rate = 'Overall rate ' + (rate * 100).toFixed(0) + '%';
```

Comment:
rate string should not be fixed here. It needs to be configurable by the `formatter` option like the existing `label.formatter` option

Reply:
Rate formatter is different with label formatter, it has to concern previous and next, so it need a new label fetcher, so sorry, I can't implement this.